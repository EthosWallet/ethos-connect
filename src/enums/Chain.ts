export enum Chain {
  Eth = "ethereum",
  Sui = "sui"
}