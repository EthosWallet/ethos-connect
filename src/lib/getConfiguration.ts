import store from 'store2'
import { EthosConfiguration } from '../types/EthosConfiguration'

const getConfiguration = (): EthosConfiguration => {
  const ethosStore = store.namespace('ethos')
  const configuration = ethosStore('configuration')
  return configuration || {}
}

export default getConfiguration
