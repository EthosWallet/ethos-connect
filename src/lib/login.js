import store from "store2";
import apiCall from "./apiCall";

const login = async (email, appId) => {
  const userStore = store.namespace('users');

  const { json: { user } } = await apiCall({
    relativePath: "users/login",
    method: "POST",
    body: { 
      email, 
      appId,
      returnTo: window.location.href
    }
  })

  userStore('current', user)
  return user;
}

export default login;