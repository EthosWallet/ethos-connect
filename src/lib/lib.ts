// import login from './login'
// import logout from './logout'
// import transact from './transact'
// import preapprove from './preapprove'
import getWalletContents from './getWalletContents'
// import showWallet from './showWallet'
// import hideWallet from './hideWallet'
// import dripSui from './dripSui'
import getConfiguration from './getConfiguration'
import getEthosSigner from './getEthosSigner'
import initializeEthos from './initializeEthos'
import postIFrameMessage from './postIFrameMessage'
import listenForMobileConnection from './listenForMobileConnection'

const lib = {
    // showWallet,
    // hideWallet,
    // login,
    // logout,
    // transact,
    // preapprove,
    getWalletContents,
    // dripSui,
    postIFrameMessage,
    getEthosSigner,
    getConfiguration,
    initializeEthos, 
    listenForMobileConnection   
}

export default lib;
