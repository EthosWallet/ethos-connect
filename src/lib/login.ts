import store from 'store2'
import { User } from 'types/User'
import getConfiguration from './getConfiguration'
import postMessage from './postMessage'

export type loginArgs = {
  email?: string, 
  provider?: string, 
  appId: string
}

const login = async ({ email, provider, appId }: loginArgs) => {
  const { walletAppUrl } = getConfiguration();
  const userStore = store.namespace('users')

  if (provider) {
    const returnTo = location.href;
    const fullUrl = `${walletAppUrl}/auth?appId=${appId}&returnTo=${returnTo}`
    location.href = `${walletAppUrl}/socialauth?redirectTo=${encodeURIComponent(fullUrl)}`;
    return;
  }

  return new Promise<User|null>((resolve, _reject) => {
    const loginEventListener = (message: any) => {
      if (message.origin === walletAppUrl) {
        const { action, data } = message.data
        if (action !== 'login') return
        window.removeEventListener('message', loginEventListener)

        console.log('LOGIN USER DATA', data)
        userStore('current', data)
        resolve(data)
      }
    }

    window.addEventListener('message', loginEventListener)

    postMessage({
      action: 'login',
      data: {
        email,
        provider,
        returnTo: window.location.href,
        appId
      },
    })
  })
}

export default login
