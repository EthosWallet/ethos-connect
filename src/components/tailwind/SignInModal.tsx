import React, { useState } from 'react'
// import connectWallet from '../../lib/connectWallet'
import login from '../../lib/login'
import WalletConnect from '../svg/WalletConnect'
import Ethos from '../svg/Ethos'
import Metamask from '../svg/Metamask'
import Loader from '..//svg/Loader'
import connectMetaMask from '../../utils/metamask'
import { ethers } from 'ethers'
// import { Signer } from '../../lib/ethersWrapper/Signer';
// import { getAppBaseUrl } from 'lib'

type SignInModalProps = {
  appId: string
  isOpen: boolean
  onClose: () => void
  setProvider: React.Dispatch<React.SetStateAction<ethers.providers.Web3Provider | undefined>>,
  setSigner: React.Dispatch<React.SetStateAction<ethers.providers.JsonRpcSigner | undefined>>
}

const SignInModal = ({ appId, isOpen, onClose, setProvider, setSigner }: SignInModalProps) => {
  const [signingIn, setSigningIn] = useState(false)
  const [email, setEmail] = useState('')

  const _login = async () => {
    // const appBaseUrl = getAppBaseUrl();
    setSigningIn(true)
    const user = await login(email, appId)
    setEmail('')
    console.log('user :>> ', user);
    if (!user) {
      setSigner(user)
    }
    // const provider = new ethers.providers.JsonRpcProvider(appBaseUrl + '/api/rpc');
    // const signer = provider.getSigner() as Signer;
    // Provider has a different type here...
    // setProvider(provider);
    // setSigner(signer);
    onClose()
  }

  const _connectWallet = async () => {
    onClose()
  }

  const _connectMetaMask = async () => {
    const { provider, signer } = await connectMetaMask();
    setProvider(provider);
    setSigner(signer);
    onClose();
  }

  const walletOptions = [
    { name: 'Ethos', logo: <Ethos />, onClick: _connectWallet },
    { name: 'Metamask', logo: <Metamask />, onClick: _connectMetaMask },
    { name: 'WalletConnect', logo: <WalletConnect />, onClick: _connectWallet },
  ]

  return (
    <div style={modalStyle(isOpen)}>
      <div style={headerStyle()}>
        <h3 style={titleStyle()}>Sign In</h3>
        <div style={closeStyle()} onClick={onClose}>
          &#x2715;
        </div>
      </div>
      <div style={mainContentStyle()}>
        <div style={walletOptionsStyle()}>
          {walletOptions.map((option, index) => (
            <div key={index} style={walletOptionStyle()} onClick={option.onClick}>
              {option.logo}
              {option.name}
            </div>
          ))}
        </div>
        <div style={registrationStyle()}>
          <h3 style={registrationHeaderStyle()}>Magic Link</h3>
          <div style={explainerStyle()}>
            Enter your email and we&#39;ll send you a magic link that will sign you in.
          </div>
          {signingIn ? (
            <div className="flex justify-center">
              <Loader width={100} />
            </div>
          ) : (
            <>
              <input
                style={inputStyle()}
                type="email"
                placeholder="Email address"
                value={email}
                onChange={(e) => setEmail(e.target.value)}
              />
              <button style={buttonStyle()} onClick={_login}>
                Send Magic Link
              </button>
            </>
          )}
        </div>
      </div>
    </div>
  )
}

const modalStyle = (isOpen: boolean) =>
({
  textAlign: 'left',
  border: '1px solid rgb(203 213 225)',
  borderRadius: '0.5rem',
  transitionProperty: 'opacity',
  transitionTimingFunction: 'cubic-bezier(0.4, 0, 0.2, 1)',
  transitionDuration: '250ms',
  opacity: isOpen ? 1 : 0,
  position: 'absolute',
  left: isOpen ? '50%' : '-9999px',
  top: '40%',
  transform: 'translate(-50%, -50%)',
  backgroundColor: 'white',
  width: '660px',
  fontWeight: '400',
} as React.CSSProperties)

const headerStyle = () =>
({
  borderBottom: '1px solid rgb(241 245 249)',
  padding: '12px',
  display: 'flex',
  justifyContent: 'space-between',
} as React.CSSProperties)

const titleStyle = () =>
({
  fontSize: '1rem',
  fontWeight: '500',
} as React.CSSProperties)

const closeStyle = () =>
({
  backgroundColor: '#F9FAFB',
  borderRadius: '100%',
  width: '24px',
  height: '24px',
  textAlign: 'center',
  color: '#A0AEBA',
  cursor: 'pointer',
} as React.CSSProperties)

const mainContentStyle = () =>
({
  display: 'flex',
  justifyContent: 'space-between',
} as React.CSSProperties)

const walletOptionsStyle = () =>
({
  width: '300px',
  padding: '24px 12px',
  borderRight: '1px solid rgb(241 245 249)',
  gap: '12px',
  display: 'flex',
  flexDirection: 'column',
} as React.CSSProperties)

const walletOptionStyle = () =>
({
  padding: '12px',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'start',
  gap: '0.75rem',
  backgroundColor: '#F9FAFB',
  borderRadius: '0.5rem',
  fontWeight: '500',
  cursor: 'pointer',
} as React.CSSProperties)

const registrationStyle = () =>
({
  padding: '24px',
  display: 'flex',
  flexDirection: 'column',
  gap: '12px',
} as React.CSSProperties)

const registrationHeaderStyle = () =>
({
  fontWeight: '500',
} as React.CSSProperties)

const explainerStyle = () =>
({
  fontSize: 'smaller',
} as React.CSSProperties)

const inputStyle = () =>
({
  border: '1px solid rgb(203 213 225)',
  borderRadius: '0.5rem',
  padding: '12px',
  width: '100%',
} as React.CSSProperties)

const buttonStyle = () =>
({
  border: '1px solid rgb(203 213 225)',
  borderRadius: '0.5rem',
  padding: '12px',
  backgroundColor: '#761AC7',
  color: '#FFFFFF',
  width: '50%',
} as React.CSSProperties)

export default SignInModal
