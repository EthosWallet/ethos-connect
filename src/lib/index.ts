import login from './login.js'
import logout from './logout.js'
import transact from './transact.js'
import requestAutomatedTransactionPermission from './requestAutomatedTransactionPermission.js'
import activeUser from './activeUser.js'
import tokenTransfers from './tokenTransfers.js'
import query from './query.js'
import getWalletBalance from './getWalletBalance.js'
import getWalletNfts from './getWalletNfts.js'
import getWalletContents from './getWalletContents.js'
import showWallet from './showWallet.js'
import hideWallet from './hideWallet.js'
import getConfiguration from './getConfiguration.js'
import initialize from './initialize.js'
import dripSui from './dripSui.js'

export {
  login,
  logout,
  transact,
  requestAutomatedTransactionPermission,
  getWalletBalance,
  getWalletNfts,
  getWalletContents,
  activeUser,
  tokenTransfers,
  query,
  showWallet,
  hideWallet,
  getConfiguration,
  initialize,
  dripSui
}
