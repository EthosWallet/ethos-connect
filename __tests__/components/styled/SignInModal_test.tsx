import React from 'react'
import { create, act } from 'react-test-renderer'

import SignInModal from '../../../src/components/styled/SignInModal'
import Ethos from '../../../src/components/svg/Ethos'
import * as lib from '../../../src/lib/login'
import * as getConfiguration from '../../../src/lib/getConfiguration'
import { Chain } from '../../../src/enums/Chain'

// import * as wagmi from 'wagmi'
import Metamask from '../../../src/components/svg/Metamask'
import WalletConnect from '../../../src/components/svg/WalletConnect'

const modalExists = (root: any) => {
  const modal = root.findByProps({ role: 'dialog' })
  return modal.props.style.display === 'block'
}

describe('SignInModal', () => {
  it('renders a hidden modal if isOpen is false', () => {
    const signInModal = create(<SignInModal isOpen={false} />)

    const root = signInModal.root
    expect(modalExists(root)).toBeFalsy()

    expect(signInModal.toJSON()).toMatchSnapshot()
  })

  it('renders a visible modal if isOpen is true', () => {
    const signInModal = create(<SignInModal isOpen={true} />)

    const root = signInModal.root
    expect(modalExists(root)).toBeTruthy()

    expect(root.findAllByType(Metamask).length).toBe(0)
    expect(root.findAllByType(WalletConnect).length).toBe(0)

    expect(signInModal.toJSON()).toMatchSnapshot()
  })

  it('shows a warning if you click the Ethos wallet button', () => {
    const warningCount = (root: any) =>
      root.findAllByProps({
        children: 'You do not have the necessary wallet extension installed.',
      }).length

    const signInModal = create(<SignInModal isOpen={true} />)

    const root = signInModal.root
    expect(warningCount(root)).toBe(0)

    const ethosWalletButton = root.findByType(Ethos)
    act(() => {
      ethosWalletButton.parent.parent.props.onClick()
    })
    expect(warningCount(root)).toBe(1)
  })

  it('sends an email if you click the send email button', async () => {
    const testEmail = 'test@example.com'

    let emailProvided
    jest.spyOn(lib, 'default').mockImplementation(({ email, provider, appId }) => {
      emailProvided = email
      expect(appId).toBe('test')
      return Promise.resolve({})
    })
    const onEmailSent = jest.fn()
    const onClose = jest.fn()

    const signInModal = create(
      <SignInModal isOpen={true} onEmailSent={onEmailSent} onClose={onClose} />
    )

    const root = signInModal.root
    const emailInput = root.findByProps({ type: 'email' })
    const emailForm = emailInput.parent

    act(() => {
      emailInput.props.onChange({ target: { value: testEmail } })
    })

    await act(async () => {
      emailForm.props.onSubmit()
    })

    expect(emailProvided).toBe(testEmail)
    expect(onEmailSent.mock.calls.length).toBe(1)
    expect(onClose.mock.calls.length).toBe(1)
  })

  // describe('Ethereum', () => {
  //   beforeEach(() => {
  //     jest.spyOn(getConfiguration, 'default').mockImplementation(() => ({
  //       walletAppUrl: 'test',
  //       appId: 'test',
  //       network: 'test',
  //       chain: Chain.Eth,
  //     }))

  //     jest.spyOn(wagmi as any, 'useConnect').mockReturnValue({
  //       connectors: [{ id: 'metaMask' }, { id: 'walletConnect' }],
  //     })
  //   })

  //   it('renders the wagmi connector buttons', () => {
  //     const signInModal = create(<SignInModal isOpen={true} />)

  //     const root = signInModal.root
  //     expect(root.findByType(Metamask)).toBeDefined()
  //     expect(root.findByType(WalletConnect)).toBeDefined()
  //     expect(signInModal.toJSON()).toMatchSnapshot()
  //   })
  // })
})
