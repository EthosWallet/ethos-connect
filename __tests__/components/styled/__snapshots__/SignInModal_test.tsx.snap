// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInModal renders a hidden modal if isOpen is false 1`] = `
<div
  role="dialog"
  style={
    Object {
      "display": "none",
      "position": "relative",
      "zIndex": "10",
    }
  }
>
  <div
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "rgb(107 114 128 / .75)",
        "bottom": "0px",
        "left": "0px",
        "position": "fixed",
        "right": "0px",
        "top": "0px",
      }
    }
  />
  <div
    style={
      Object {
        "bottom": "0px",
        "left": "0px",
        "overflowY": "auto",
        "position": "fixed",
        "right": "0px",
        "top": "0px",
        "zIndex": "99",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
          "minHeight": "100%",
          "padding": "0",
          "textAlign": "center",
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderRadius": "0.5rem",
            "boxShadow": "0 20px 25px -5px rgba(0, 0, 0, 0.1), 0 10px 10px -5px rgba(0, 0, 0, 0.04)",
            "maxWidth": "400px",
            "overflow": "hidden",
            "position": "relative",
            "transitionProperty": "all",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "height": "36px",
            }
          }
        >
          <span
            style={
              Object {
                "backgroundColor": "#F9FAFB",
                "borderRadius": "100%",
                "color": "#A0AEBA",
                "cursor": "pointer",
                "float": "right",
                "height": "24px",
                "marginRight": "12px",
                "marginTop": "12px",
                "textAlign": "center",
                "width": "24px",
              }
            }
          >
            âœ•
          </span>
        </div>
        <div
          style={
            Object {
              "padding": "0 60px 0 60px",
            }
          }
        >
          <span
            style={
              Object {
                "color": "#999999",
                "fontSize": "12px",
                "lineHeight": "24px",
              }
            }
          >
            Sign in to
          </span>
          <h2>
            <span
              style={
                Object {
                  "display": "inline-flex",
                }
              }
            >
              <svg
                fill="none"
                height={27.659574468085108}
                viewBox="0 0 47 65"
                width={20}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M6.00471 1H40.0029C42.7644 1 45.0029 3.23858 45.0029 6V44.8425C45.0029 47.604 42.7643 49.8425 40.0029 49.8425H6.0047C3.24328 49.8425 1.0047 47.604 1.0047 44.8425V6C1.0047 3.23858 3.24329 1 6.00471 1Z"
                  stroke="#1e293b"
                  strokeWidth="2"
                />
                <path
                  d="M6.68764 3.64648L30.6631 14.8026C32.0736 15.4589 32.9756 16.8735 32.9756 18.4292V58.6799C32.9756 61.5743 29.9966 63.5105 27.3515 62.3353L3.37601 51.683C1.93126 51.0411 1.00013 49.6085 1.00013 48.0276V7.27309C1.00013 4.34854 4.03609 2.41268 6.68764 3.64648Z"
                  fill="#1e293b"
                />
              </svg>
              <span
                style={
                  Object {
                    "fontSize": "18px",
                    "fontWeight": "600",
                    "lineHeight": "24px",
                    "marginLeft": "10px",
                  }
                }
              >
                Ethos Wallet
              </span>
            </span>
          </h2>
          <div
            style={
              Object {
                "marginTop": "16px",
              }
            }
          >
            <span
              style={
                Object {
                  "fontSize": "14px",
                  "fontWeight": "600",
                }
              }
            >
              Sign in with your email
            </span>
          </div>
          <form
            onSubmit={[Function]}
          >
            <input
              onChange={[Function]}
              placeholder="Email address"
              style={
                Object {
                  "border": "1px solid rgb(203 213 225)",
                  "borderRadius": "0.5rem",
                  "marginTop": "12px",
                  "padding": "12px",
                  "width": "100%",
                }
              }
              type="email"
              value=""
            />
            <button
              style={
                Object {
                  "backgroundColor": "#761AC7",
                  "border": "1px solid rgb(203 213 225)",
                  "borderRadius": "0.5rem",
                  "color": "#FFFFFF",
                  "fontWeight": "600",
                  "marginTop": ".75rem",
                  "padding": "12px 20px 12px 20px",
                  "textDecoration": "none",
                  "width": "100%",
                }
              }
              type="submit"
            >
              Sign In
            </button>
          </form>
          <div
            style={
              Object {
                "margin": "16px 0 16px 0",
              }
            }
          >
            <span
              style={
                Object {
                  "color": "#999999",
                  "fontSize": "12px",
                  "lineHeight": "24px",
                }
              }
            >
              or
            </span>
          </div>
          <span
            style={
              Object {
                "fontSize": "14px",
                "fontWeight": "600",
              }
            }
          >
            Connect an existing wallet
          </span>
          <div
            style={
              Object {
                "alignItems": "flex-start",
                "display": "flex",
                "flexDirection": "column",
                "gap": "8px",
                "marginBottom": "32px",
                "marginTop": "12px",
                "padding": "0px",
              }
            }
          >
            <button
              onClick={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "background": "#FFFFFF",
                  "border": "1px solid #E9EBED",
                  "borderRadius": "8px",
                  "boxShadow": "0px 2px 8px rgba(32, 39, 44, 0.08), 0px 0px 1px rgba(32, 39, 44, 0.08)",
                  "boxSizing": "border-box",
                  "display": "flex",
                  "flex": "none",
                  "flexDirection": "row",
                  "flexGrow": "0",
                  "gap": "10px",
                  "height": "48px",
                  "justifyContent": "center",
                  "order": "0",
                  "padding": "12px 20px",
                  "width": "100%",
                }
              }
            >
              <span
                style={
                  Object {
                    "display": "inline-flex",
                  }
                }
              >
                <svg
                  fill="none"
                  height={23.51063829787234}
                  viewBox="0 0 47 65"
                  width={17}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6.00471 1H40.0029C42.7644 1 45.0029 3.23858 45.0029 6V44.8425C45.0029 47.604 42.7643 49.8425 40.0029 49.8425H6.0047C3.24328 49.8425 1.0047 47.604 1.0047 44.8425V6C1.0047 3.23858 3.24329 1 6.00471 1Z"
                    stroke="#5B5D5F"
                    strokeWidth="2"
                  />
                  <path
                    d="M6.68764 3.64648L30.6631 14.8026C32.0736 15.4589 32.9756 16.8735 32.9756 18.4292V58.6799C32.9756 61.5743 29.9966 63.5105 27.3515 62.3353L3.37601 51.683C1.93126 51.0411 1.00013 49.6085 1.00013 48.0276V7.27309C1.00013 4.34854 4.03609 2.41268 6.68764 3.64648Z"
                    fill="#5B5D5F"
                  />
                </svg>
                <span
                  style={
                    Object {
                      "color": "#5B5D5F",
                      "marginLeft": "10px",
                    }
                  }
                >
                  Ethos Wallet
                </span>
              </span>
            </button>
            <button
              onClick={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "background": "#FFFFFF",
                  "border": "1px solid #E9EBED",
                  "borderRadius": "8px",
                  "boxShadow": "0px 2px 8px rgba(32, 39, 44, 0.08), 0px 0px 1px rgba(32, 39, 44, 0.08)",
                  "boxSizing": "border-box",
                  "display": "flex",
                  "flex": "none",
                  "flexDirection": "row",
                  "flexGrow": "0",
                  "gap": "10px",
                  "height": "48px",
                  "justifyContent": "center",
                  "order": "0",
                  "padding": "12px 20px",
                  "width": "100%",
                }
              }
            >
              <span
                style={
                  Object {
                    "display": "inline-flex",
                  }
                }
              >
                <svg
                  height={21.42857142857143}
                  id="SuiSvg"
                  viewBox="-1 0 28 40"
                  width={15}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1.8611,33.0541a13.6477,13.6477,0,0,0,23.7778,0,13.89,13.89,0,0,0,0-13.8909L15.1824.8368a1.6444,1.6444,0,0,0-2.8648,0L1.8611,19.1632A13.89,13.89,0,0,0,1.8611,33.0541ZM10.8044,9.5555,13.0338,5.648a.8222.8222,0,0,1,1.4324,0L23.043,20.68a10.8426,10.8426,0,0,1,.8873,8.8828,9.4254,9.4254,0,0,0-.4388-1.4586c-1.1847-3.0254-3.8634-5.36-7.9633-6.9393-2.8187-1.0819-4.618-2.6731-5.3491-4.73C9.2375,13.7848,10.221,10.8942,10.8044,9.5555ZM7.0028,16.2184,4.457,20.68a10.8569,10.8569,0,0,0,0,10.8582,10.6776,10.6776,0,0,0,16.1566,2.935,7.5061,7.5061,0,0,0,.0667-5.2913c-.87-2.1858-2.9646-3.9308-6.2252-5.1876-3.6857-1.4147-6.08-3.6233-7.1157-6.5625A9.297,9.297,0,0,1,7.0028,16.2184Z"
                    style={
                      Object {
                        "fill": "#5B5D5F",
                        "fillRule": "evenodd",
                      }
                    }
                  />
                </svg>
                <span
                  style={
                    Object {
                      "color": "#5B5D5F",
                      "marginLeft": "10px",
                    }
                  }
                >
                  Sui Wallet
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SignInModal renders a visible modal if isOpen is true 1`] = `
<div
  role="dialog"
  style={
    Object {
      "display": "block",
      "position": "relative",
      "zIndex": "10",
    }
  }
>
  <div
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "rgb(107 114 128 / .75)",
        "bottom": "0px",
        "left": "0px",
        "position": "fixed",
        "right": "0px",
        "top": "0px",
      }
    }
  />
  <div />
  <div
    style={
      Object {
        "bottom": "0px",
        "left": "0px",
        "overflowY": "auto",
        "position": "fixed",
        "right": "0px",
        "top": "0px",
        "zIndex": "99",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
          "minHeight": "100%",
          "padding": "0",
          "textAlign": "center",
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderRadius": "0.5rem",
            "boxShadow": "0 20px 25px -5px rgba(0, 0, 0, 0.1), 0 10px 10px -5px rgba(0, 0, 0, 0.04)",
            "maxWidth": "400px",
            "overflow": "hidden",
            "position": "relative",
            "transitionProperty": "all",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "height": "36px",
            }
          }
        >
          <span
            style={
              Object {
                "backgroundColor": "#F9FAFB",
                "borderRadius": "100%",
                "color": "#A0AEBA",
                "cursor": "pointer",
                "float": "right",
                "height": "24px",
                "marginRight": "12px",
                "marginTop": "12px",
                "textAlign": "center",
                "width": "24px",
              }
            }
          >
            âœ•
          </span>
        </div>
        <div
          style={
            Object {
              "padding": "0 60px 0 60px",
            }
          }
        >
          <span
            style={
              Object {
                "color": "#999999",
                "fontSize": "12px",
                "lineHeight": "24px",
              }
            }
          >
            Sign in to
          </span>
          <h2>
            <span
              style={
                Object {
                  "display": "inline-flex",
                }
              }
            >
              <svg
                fill="none"
                height={27.659574468085108}
                viewBox="0 0 47 65"
                width={20}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M6.00471 1H40.0029C42.7644 1 45.0029 3.23858 45.0029 6V44.8425C45.0029 47.604 42.7643 49.8425 40.0029 49.8425H6.0047C3.24328 49.8425 1.0047 47.604 1.0047 44.8425V6C1.0047 3.23858 3.24329 1 6.00471 1Z"
                  stroke="#1e293b"
                  strokeWidth="2"
                />
                <path
                  d="M6.68764 3.64648L30.6631 14.8026C32.0736 15.4589 32.9756 16.8735 32.9756 18.4292V58.6799C32.9756 61.5743 29.9966 63.5105 27.3515 62.3353L3.37601 51.683C1.93126 51.0411 1.00013 49.6085 1.00013 48.0276V7.27309C1.00013 4.34854 4.03609 2.41268 6.68764 3.64648Z"
                  fill="#1e293b"
                />
              </svg>
              <span
                style={
                  Object {
                    "fontSize": "18px",
                    "fontWeight": "600",
                    "lineHeight": "24px",
                    "marginLeft": "10px",
                  }
                }
              >
                Ethos Wallet
              </span>
            </span>
          </h2>
          <div
            style={
              Object {
                "marginTop": "16px",
              }
            }
          >
            <span
              style={
                Object {
                  "fontSize": "14px",
                  "fontWeight": "600",
                }
              }
            >
              Sign in with your email
            </span>
          </div>
          <form
            onSubmit={[Function]}
          >
            <input
              onChange={[Function]}
              placeholder="Email address"
              style={
                Object {
                  "border": "1px solid rgb(203 213 225)",
                  "borderRadius": "0.5rem",
                  "marginTop": "12px",
                  "padding": "12px",
                  "width": "100%",
                }
              }
              type="email"
              value=""
            />
            <button
              style={
                Object {
                  "backgroundColor": "#761AC7",
                  "border": "1px solid rgb(203 213 225)",
                  "borderRadius": "0.5rem",
                  "color": "#FFFFFF",
                  "fontWeight": "600",
                  "marginTop": ".75rem",
                  "padding": "12px 20px 12px 20px",
                  "textDecoration": "none",
                  "width": "100%",
                }
              }
              type="submit"
            >
              Sign In
            </button>
          </form>
          <div
            style={
              Object {
                "margin": "16px 0 16px 0",
              }
            }
          >
            <span
              style={
                Object {
                  "color": "#999999",
                  "fontSize": "12px",
                  "lineHeight": "24px",
                }
              }
            >
              or
            </span>
          </div>
          <span
            style={
              Object {
                "fontSize": "14px",
                "fontWeight": "600",
              }
            }
          >
            Connect an existing wallet
          </span>
          <div
            style={
              Object {
                "alignItems": "flex-start",
                "display": "flex",
                "flexDirection": "column",
                "gap": "8px",
                "marginBottom": "32px",
                "marginTop": "12px",
                "padding": "0px",
              }
            }
          >
            <button
              onClick={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "background": "#FFFFFF",
                  "border": "1px solid #E9EBED",
                  "borderRadius": "8px",
                  "boxShadow": "0px 2px 8px rgba(32, 39, 44, 0.08), 0px 0px 1px rgba(32, 39, 44, 0.08)",
                  "boxSizing": "border-box",
                  "display": "flex",
                  "flex": "none",
                  "flexDirection": "row",
                  "flexGrow": "0",
                  "gap": "10px",
                  "height": "48px",
                  "justifyContent": "center",
                  "order": "0",
                  "padding": "12px 20px",
                  "width": "100%",
                }
              }
            >
              <span
                style={
                  Object {
                    "display": "inline-flex",
                  }
                }
              >
                <svg
                  fill="none"
                  height={23.51063829787234}
                  viewBox="0 0 47 65"
                  width={17}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6.00471 1H40.0029C42.7644 1 45.0029 3.23858 45.0029 6V44.8425C45.0029 47.604 42.7643 49.8425 40.0029 49.8425H6.0047C3.24328 49.8425 1.0047 47.604 1.0047 44.8425V6C1.0047 3.23858 3.24329 1 6.00471 1Z"
                    stroke="#5B5D5F"
                    strokeWidth="2"
                  />
                  <path
                    d="M6.68764 3.64648L30.6631 14.8026C32.0736 15.4589 32.9756 16.8735 32.9756 18.4292V58.6799C32.9756 61.5743 29.9966 63.5105 27.3515 62.3353L3.37601 51.683C1.93126 51.0411 1.00013 49.6085 1.00013 48.0276V7.27309C1.00013 4.34854 4.03609 2.41268 6.68764 3.64648Z"
                    fill="#5B5D5F"
                  />
                </svg>
                <span
                  style={
                    Object {
                      "color": "#5B5D5F",
                      "marginLeft": "10px",
                    }
                  }
                >
                  Ethos Wallet
                </span>
              </span>
            </button>
            <button
              onClick={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "background": "#FFFFFF",
                  "border": "1px solid #E9EBED",
                  "borderRadius": "8px",
                  "boxShadow": "0px 2px 8px rgba(32, 39, 44, 0.08), 0px 0px 1px rgba(32, 39, 44, 0.08)",
                  "boxSizing": "border-box",
                  "display": "flex",
                  "flex": "none",
                  "flexDirection": "row",
                  "flexGrow": "0",
                  "gap": "10px",
                  "height": "48px",
                  "justifyContent": "center",
                  "order": "0",
                  "padding": "12px 20px",
                  "width": "100%",
                }
              }
            >
              <span
                style={
                  Object {
                    "display": "inline-flex",
                  }
                }
              >
                <svg
                  height={21.42857142857143}
                  id="SuiSvg"
                  viewBox="-1 0 28 40"
                  width={15}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1.8611,33.0541a13.6477,13.6477,0,0,0,23.7778,0,13.89,13.89,0,0,0,0-13.8909L15.1824.8368a1.6444,1.6444,0,0,0-2.8648,0L1.8611,19.1632A13.89,13.89,0,0,0,1.8611,33.0541ZM10.8044,9.5555,13.0338,5.648a.8222.8222,0,0,1,1.4324,0L23.043,20.68a10.8426,10.8426,0,0,1,.8873,8.8828,9.4254,9.4254,0,0,0-.4388-1.4586c-1.1847-3.0254-3.8634-5.36-7.9633-6.9393-2.8187-1.0819-4.618-2.6731-5.3491-4.73C9.2375,13.7848,10.221,10.8942,10.8044,9.5555ZM7.0028,16.2184,4.457,20.68a10.8569,10.8569,0,0,0,0,10.8582,10.6776,10.6776,0,0,0,16.1566,2.935,7.5061,7.5061,0,0,0,.0667-5.2913c-.87-2.1858-2.9646-3.9308-6.2252-5.1876-3.6857-1.4147-6.08-3.6233-7.1157-6.5625A9.297,9.297,0,0,1,7.0028,16.2184Z"
                    style={
                      Object {
                        "fill": "#5B5D5F",
                        "fillRule": "evenodd",
                      }
                    }
                  />
                </svg>
                <span
                  style={
                    Object {
                      "color": "#5B5D5F",
                      "marginLeft": "10px",
                    }
                  }
                >
                  Sui Wallet
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
